src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1261,35 +1261,65 @@ private void Init()
 
 		private ICurrentSessionContext BuildCurrentSessionContext()
 		{
```

Comment:
I would not change this method and would not change the standard contexts' constructors. Also I would make the `ICurrentSessionContextWithFactory` interface optional.

Reply:
If this method is not changed then `ICurrentSessionContextWithFactory` would not have any usage and should not be kept.

So, do you mean you would not change just the direct instantiation cases?

About making `ICurrentSessionContextWithFactory` optional, then the obsoleted line:
```c#
(ICurrentSessionContext) Environment.ObjectsFactory.CreateInstance(implClass, this);
```

Would have to be replaced by:
```c#